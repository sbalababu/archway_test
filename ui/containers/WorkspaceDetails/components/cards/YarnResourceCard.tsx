import * as React from 'react';
import { Card, Row, Col, Button, Spin } from 'antd';

import { ResourcePool } from '../../../../models/Workspace';
import { ModalType } from '../../../../constants';

interface Props {
  resource: ResourcePool;
  resourcePoolLoading: boolean;
  showModal: (e: React.MouseEvent, type: ModalType) => void;
}

const YarnResourceCard = ({ resource, resourcePoolLoading, showModal }: Props) => (
  <Card bordered bodyStyle={{ padding: 8 }}>
    <Row style={{ textAlign: 'center', height: 150 }} type="flex" align="middle">
      <Col span={24}>YARN RESOURCES</Col>
      {resourcePoolLoading ? (
        <div style={{ width: '100%', textAlign: 'center' }}>
          <Spin tip="Updating" />
        </div>
      ) : (
        <React.Fragment>
          <Col span={24}>
            <b style={{ fontSize: 12 }}>{resource.pool_name}</b>
            <br />
            QUEUE NAME
          </Col>
          <Col span={12}>
            <b>{resource.max_cores}</b>
            <br />
            MAX CORES
          </Col>
          <Col span={12}>
            <b>{resource.max_memory_in_gb}GB</b>
            <br />
            MAX MEMORY
          </Col>
          <Col span={24}>
            <Button type="primary" onClick={e => showModal(e, ModalType.ModifyCoreMemory)}>
              Update
            </Button>
          </Col>
        </React.Fragment>
      )}
    </Row>
  </Card>
);

export default YarnResourceCard;
