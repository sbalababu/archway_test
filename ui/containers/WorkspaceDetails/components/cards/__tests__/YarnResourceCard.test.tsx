import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import YarnResourceCard from '../YarnResourceCard';

describe('YarnResourceCard', () => {
  it('renders correctly', () => {
    const props = {
      resource: {
        id: 1,
        pool_name: 'pool',
        max_cores: 4,
        max_memory_in_gb: 4,
      },
      resourcePoolLoading: false,
      showModal: () => null,
    };
    const wrapper = shallow(<YarnResourceCard {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
