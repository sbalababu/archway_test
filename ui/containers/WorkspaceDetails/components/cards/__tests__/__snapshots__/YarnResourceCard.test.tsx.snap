// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YarnResourceCard renders correctly 1`] = `
<Card
  bodyStyle={
    Object {
      "padding": 8,
    }
  }
  bordered={true}
>
  <Row
    align="middle"
    gutter={0}
    style={
      Object {
        "height": 150,
        "textAlign": "center",
      }
    }
    type="flex"
  >
    <Col
      span={24}
    >
      YARN RESOURCES
    </Col>
    <Col
      span={24}
    >
      <b
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        pool
      </b>
      <br />
      QUEUE NAME
    </Col>
    <Col
      span={12}
    >
      <b>
        4
      </b>
      <br />
      MAX CORES
    </Col>
    <Col
      span={12}
    >
      <b>
        4
        GB
      </b>
      <br />
      MAX MEMORY
    </Col>
    <Col
      span={24}
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Update
      </Button>
    </Col>
  </Row>
</Card>
`;
